UserData expected `*mut u8`, found `*mut i8`

I am trying to use `russimp` with OpenGL, but when I add it to my project and try to build I get the following error:
```
error[E0308]: mismatched types
  --> /home/pi/.cargo/registry/src/index.crates.io-6f17d22bba15001f/russimp-3.2.0/src/fs.rs:49:27
   |
49 |                 UserData: user_data,
   |                           ^^^^^^^^^ expected `*mut u8`, found `*mut i8`
   |
   = note: expected raw pointer `*mut u8`
              found raw pointer `*mut i8`

error[E0308]: mismatched types
  --> /home/pi/.cargo/registry/src/index.crates.io-6f17d22bba15001f/russimp-3.2.0/src/fs.rs:91:23
   |
91 |             UserData: user_data,
   |                       ^^^^^^^^^ expected `*mut u8`, found `*mut i8`
   |
   = note: expected raw pointer `*mut u8`
              found raw pointer `*mut i8`
```
Any help would be greatly appreciated.

--- a/src/fs.rs
+++ b/src/fs.rs
@@ -41,9 +41,9 @@ impl<T: FileSystem> FileOperationsWrapper<T> {
         let trait_obj: &dyn FileSystem = file_system;
         let managed_box = Box::new(trait_obj);
         let user_data = Box::into_raw(managed_box);
-        #[cfg(not(any(target_arch = "arm", target_arch = "aarch64")))]
+        #[cfg(any(target_arch = "x86", target_arch = "x86_64", target_os = "macos"))]
         let user_data = user_data as *mut i8;
-        #[cfg(any(target_arch = "arm", target_arch = "aarch64"))]
+        #[cfg(all(any(target_arch = "arm", target_arch = "aarch64"), not(target_os = "macos")))]
         let user_data = user_data as *mut u8;
         FileOperationsWrapper {
             ai_file: aiFileIO {
@@ -83,9 +83,9 @@ impl<T: FileSystem> FileOperationsWrapper<T> {
         // raw pointer that can be stuffed in the UserData.
         let double_box = Box::new(file);
         let managed_box = Box::into_raw(double_box); // Cleaned up in io_close.
-        #[cfg(not(any(target_arch = "arm", target_arch = "aarch64")))]
+        #[cfg(any(target_arch = "x86", target_arch = "x86_64", target_os = "macos"))]
         let user_data = managed_box as *mut i8;
-        #[cfg(any(target_arch = "arm", target_arch = "aarch64"))]
+        #[cfg(all(any(target_arch = "arm", target_arch = "aarch64"), not(target_os = "macos")))]
         let user_data = managed_box as *mut u8;
         let ai_file = aiFile {
             ReadProc: Some(Self::io_read),

--- a/src/fs.rs
+++ b/src/fs.rs
@@ -41,7 +41,7 @@ impl<T: FileSystem> FileOperationsWrapper<T> {
         let trait_obj: &dyn FileSystem = file_system;
         let managed_box = Box::new(trait_obj);
         let user_data = Box::into_raw(managed_box);
-        let user_data = user_data as *mut i8;
+        let user_data = user_data as *mut u8;
         FileOperationsWrapper {
             ai_file: aiFileIO {
                 OpenProc: Some(FileOperationsWrapper::<T>::io_open),
@@ -80,7 +80,7 @@ impl<T: FileSystem> FileOperationsWrapper<T> {
         // raw pointer that can be stuffed in the UserData.
         let double_box = Box::new(file);
         let managed_box = Box::into_raw(double_box); // Cleaned up in io_close.
-        let user_data = managed_box as *mut i8;
+        let user_data = managed_box as *mut u8;
         let ai_file = aiFile {
             ReadProc: Some(Self::io_read),
             WriteProc: Some(Self::io_write),

--- a/src/fs.rs
+++ b/src/fs.rs
@@ -41,6 +41,9 @@ impl<T: FileSystem> FileOperationsWrapper<T> {
         let trait_obj: &dyn FileSystem = file_system;
         let managed_box = Box::new(trait_obj);
         let user_data = Box::into_raw(managed_box);
+        #[cfg(not(any(target_arch = "arm", target_arch = "aarch64")))]
+        let user_data = user_data as *mut i8;
+        #[cfg(any(target_arch = "arm", target_arch = "aarch64"))]
         let user_data = user_data as *mut u8;
         FileOperationsWrapper {
             ai_file: aiFileIO {
@@ -80,6 +83,9 @@ impl<T: FileSystem> FileOperationsWrapper<T> {
         // raw pointer that can be stuffed in the UserData.
         let double_box = Box::new(file);
         let managed_box = Box::into_raw(double_box); // Cleaned up in io_close.
+        #[cfg(not(any(target_arch = "arm", target_arch = "aarch64")))]
+        let user_data = managed_box as *mut i8;
+        #[cfg(any(target_arch = "arm", target_arch = "aarch64"))]
         let user_data = managed_box as *mut u8;
         let ai_file = aiFile {
             ReadProc: Some(Self::io_read),

